tidy3d/components/geometry/base.py

```diff
@@ -44,6 +40,26 @@
 
 POLY_GRID_SIZE = 1e-12
 
+try:
+    trimesh_available = True
+    import trimesh
+except ImportError:
+    trimesh_available = False
+
+
+def requires_trimesh(fn):
```

Comment:
add docstrings? "when decorating a method, requires that trimesh is importable" or something.